### DISCUSSION TITLE
Can you please go easy on the deprecation warnings?

### DISCUSSION THREAD

**Discussion by sdbbs** [colour-science/colour#867] [Ideas]
Not really a bug, just wanted to point out this:

I am using colour-science `colour.__version__` of 0.3.16.

So, at a point in my code, I have:

    cctf_encoding=colour.sRGB_COLOURSPACE.cctf_encoding,

When I run my code, I get:

> .../Jupyter/notebook/lib/python3.8/site-packages/colour/utilities/verbose.py:237: ColourUsageWarning: "colour.sRGB_COLOURSPACE" object access is deprecated and will change to "colour.models.RGB_COLOURSPACE_sRGB" in a future release.
>  warn(*args, **kwargs)

Sure, no problem - let me then just change to this:

    cctf_encoding=colour.RGB_COLOURSPACE_sRGB.cctf_encoding,

... but now I get:

> AttributeError: module 'colour' has no attribute 'RGB_COLOURSPACE_sRGB'

Eh ... I thought I could get rid of the warning by:

    colour.RGB_COLOURSPACE_sRGB = colour.sRGB_COLOURSPACE

... but no, warning recognizes it is the "deprecated" access, and is raised again. Filtering it finally works:

```python
import warnings
warnings.filterwarnings("ignore", message='"colour.sRGB_COLOURSPACE" object access is deprecated and will change to')
```

So, I'd like to suggest, to only enable deprecation warnings, if one can do something about them - since it looks that for now, I will just have to filter this warning.

But it's not that big of an issue for me - so let me just close with: Thanks for a great library!

**Comment by thomasmansencal** [comment]
Hello @sdbbs,

So, when you did `cctf_encoding=colour.RGB_COLOURSPACE_sRGB.cctf_encoding`, there is a missing `.models` part that is indicated in the warning. Your access path should effectively now be: `cctf_encoding=colour.modelsRGB_COLOURSPACE_sRGB.cctf_encoding`

After we deprecate something, you should really not do that `colour.RGB_COLOURSPACE_sRGB = colour.sRGB_COLOURSPACE` but adopt the new access path or create a new one with the new path (not recommended): `colour.RGB_COLOURSPACE_sRGB = colour.models.RGB_COLOURSPACE_sRGB`. The reason is because `colour.sRGB_COLOURSPACE` will disappear in a future release!

```python
>>> import colour
>>> colour.sRGB_COLOURSPACE
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kelsolaar/Documents/Development/colour-science/colour/colour/__init__.py", line 383, in __getattr__
    return super(colour, self).__getattr__(attribute)
  File "/Users/kelsolaar/Documents/Development/colour-science/colour/colour/utilities/deprecation.py", line 349, in __getattr__
    return getattr(self._module, attribute)
AttributeError: module 'colour' has no attribute 'sRGB_COLOURSPACE'
>>> colour.RGB_COLOURSPACE_sRGB
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kelsolaar/Documents/Development/colour-science/colour/colour/__init__.py", line 383, in __getattr__
    return super(colour, self).__getattr__(attribute)
  File "/Users/kelsolaar/Documents/Development/colour-science/colour/colour/utilities/deprecation.py", line 349, in __getattr__
    return getattr(self._module, attribute)
AttributeError: module 'colour' has no attribute 'RGB_COLOURSPACE_sRGB'
>>> colour.models.RGB_COLOURSPACE_sRGB
RGB_Colourspace(sRGB,
                [[ 0.64,  0.33],
                 [ 0.3 ,  0.6 ],
                 [ 0.15,  0.06]],
                [ 0.3127,  0.329 ],
                D65,
                [[ 0.4124,  0.3576,  0.1805],
                 [ 0.2126,  0.7152,  0.0722],
                 [ 0.0193,  0.1192,  0.9505]],
                [[ 3.2406, -1.5372, -0.4986],
                 [-0.9689,  1.8758,  0.0415],
                 [ 0.0557, -0.204 ,  1.057 ]],
                <function eotf_inverse_sRGB at 0x1297be940>,
                <function eotf_sRGB at 0x1297be9d0>,
                False,
                False)
>>> colour.RGB_COLOURSPACE_sRGB = colour.models.RGB_COLOURSPACE_sRGB
>>> colour.RGB_COLOURSPACE_sRGB
RGB_Colourspace(sRGB,
                [[ 0.64,  0.33],
                 [ 0.3 ,  0.6 ],
                 [ 0.15,  0.06]],
                [ 0.3127,  0.329 ],
                D65,
                [[ 0.4124,  0.3576,  0.1805],
                 [ 0.2126,  0.7152,  0.0722],
                 [ 0.0193,  0.1192,  0.9505]],
                [[ 3.2406, -1.5372, -0.4986],
                 [-0.9689,  1.8758,  0.0415],
                 [ 0.0557, -0.204 ,  1.057 ]],
                <function eotf_inverse_sRGB at 0x1297be940>,
                <function eotf_sRGB at 0x1297be9d0>,
                False,
                False)
```
